### PR TITLE
Chk 1963 setup wallet mongodb

### PR DESCRIPTION
**Pull request by ciuffagianluca**
Create wallet resources for collection storage

### List of changes

Add configuration for database cosmodb and standard alert (as done for ecommerce)

### Motivation and context

This PR has the goal of providing the cosmodb resources and alert for wallet-service

### Type of changes

- [x] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### CODE REVIEW

File: src/domains/wallet-common/03_cosmosdb.tf [hunk 1]
```diff
@@ -74,3 +74,124 @@ resource "azurerm_cosmosdb_mongo_database" "wallet" {
...
+      indexes = [{
+          keys   = ["_id"]
+          unique = true
+        },
+        {
+          keys   = ["walletId", "eventType", "timestamp"]
+          unique = true
+        },
+        {
+          keys   = ["walletId"]
```
Comment (pietro-tota) [inline src/domains/wallet-common/03_cosmosdb.tf:116]: this index should be already covered by the above one, so maybe it's useless. What do you think?

Comment (ciuffagianluca) [reply src/domains/wallet-common/03_cosmosdb.tf:116]: We updated the indexes composition and definition. Thank you

File: src/domains/wallet-common/03_cosmosdb.tf [hunk 2]
```diff
@@ -74,3 +74,120 @@ resource "azurerm_cosmosdb_mongo_database" "wallet" {
...
+
+  name                = each.value.name
+  resource_group_name = azurerm_resource_group.cosmosdb_wallet_rg[0].name
+
+  cosmosdb_mongo_account_name  = module.cosmosdb_account_mongodb[0].name
+  cosmosdb_mongo_database_name = azurerm_cosmosdb_mongo_database.wallet[0].name
+
+  indexes     = each.value.indexes
+  shard_key   = each.value.shard_key
+  lock_enable = var.env_short == "d" ? false : true
```
Comment (infantesimone) [inline src/domains/wallet-common/03_cosmosdb.tf:139]: since we are refining the models/db, it is better if we enable the lock only in PROD

**Review by infantesimone — CHANGES_REQUESTED** [review]

File: src/domains/wallet-common/03_cosmosdb.tf (hunk 2 above)
Comment (ciuffagianluca) [reply src/domains/wallet-common/03_cosmosdb.tf:139]: You're right!

**Review by infantesimone — APPROVED** [review]

**Pull request merged**